docs/remote-feature-flags.md

```diff
@@ -0,0 +1,171 @@
+# Remote Feature Flags
+
+## Overview
+
+This document outlines the process for adding and managing remote feature flags in MetaMask (Extension and Mobile). Remote feature flags allow us to control feature availability and behavior in production without requiring a new release. They are created on the Launch Darkly platform and distributed by our internal API [ClientConfigAPI](https://github.com/consensys-vertical-apps/mmwp-client-config-api). Please read the [Remote feature flags ADR](https://github.com/MetaMask/decisions/pull/43) for more details.
+
```

Comment:
Maybe we can add a short introduction explaining what kind feature flags people should use, and redirect them to the right section:
- If they only want the ability to turn a feature ON/OF, they shall use a `Boolean flag`
- If they want to control the % rollout of a feature, they shall use a `Object flag with scope based on "threshold"`
- If they want to A/B test feature, they shall use a `Object flag with scope based on "threshold"`

Reply:
I think it's a great idea to include this section for dev who's not familiar with the types, it also matches the analysis from ADR. Addressed d097ff0f7dd9b7c5908abf7c2e07982e4c04165d

Reply:
Nice